src/project/network/Group.java

```diff
@@ -21,6 +24,26 @@
         consumption = power;
         originalconsumption=power;
     }
+    
+    /**
+     * Constructeur. Version avec la variation de consommation,Assignation du gestionnaire & du type
+     * @param power la puissance consommée par le groupe.
+     * @param s le nom du groupe.
+     */
+    Group(int power, String s,ConsumptionMacro consumMacro, ConsumptionType consumpType){
+        this(power,s);
+        this.consumpMacro=consumMacro;
+        this.consumpType=consumpType;
+    }
+    /**
+     * Constructeur. Version avec la variation de consommation,Assignation du gestionnaire & du type
+     * @param power la puissance consommée par le groupe.
+     * @param s le nom du groupe.
+     */
+    Group(int power, String s,ConsumptionType consumpType){
+        this(power,s);
+        this.consumpType=consumpType;
```

Comment:
Du coup je ne sais pas si consumMacro est supposé être statique ou non mais je pense que non sinon tu ne le donnerais pas au constructeur précédent et tu n'aurais pas écrit `this.consumpMacro`, mais du coup en supposant que j'ai raison à quoi sert un groupe qui n'a qu'un _consumpType_ et pas de _consumMacro_ ?

Reply:
En fait moi je vois ConsumpMacro comme un membre statique,unique pour tous les Group.
Après c'est plus un manque de savoir de ma part sur la bonne procédure à faire vis à vis de ça.
Du coup ce constructeur devrait être celui tout le temps utiliser une fois que le Con.Macro a été assigné, tu peux voir dans le test que le groupe 2 utilise ce constructeur.

Reply:
Tu devrais plutôt rendre consumpMacro accessible ou écrire une méthode statique pour le modifier, ce serait mieux que faire ça dans un constructeur spécial.